pkg/cmd/cli/cmd/set/volume.go

```diff
@@ -237,10 +237,6 @@ func (v *VolumeOptions) Validate(args []string) error {
 
 func (a *AddVolumeOptions) Validate(isAddOp bool) error {
 	if isAddOp {
-		if len(a.MountPath) == 0 {
```

Comment:
> if we don't have the info to validate it here, and there are valid cases (like overwrite) where a path is not required, move the check into RunVolume where we can distinguish between the two cases

According to @smarterclayton sometimes you want to add a volume without having a mount-path for it.  I'm not sure what for, but that makes the add and update cases indistinguishable.